File: IterativePostOrderTraversal2Stack.java
import java.util.*;
import java.io.*;

class Solution{
    public List<Integer> postorderTraversal(TreeNode root){
        stack<TreeNode> s1 = new Stack<TreeNode>();
        stack<TreeNode> s2 = new Stack<TreeNode>();
        List<Integer> postOrder = new ArrayList<Integer>();

        if(root == null)
            return postOrder;
        
        s1.push(root);
        while(!s1.isEmpty()){
            root= s1.pop();
            s2.add(root);
            if(root.left!=null) s1.push(root.left);
            if(root.right!=null) s1.push(root.right);
        }
        while(!s2.isEmpty()){
            postOrder.add(s2.pop().val);
        }
        return postOrder;
    }
}

File: IterativePostOrderTraversal2Stack.cpp
#include<iostream>
using namespace std;

class Solution{
    public:
        vector<int> postOrderTraversal(TreeNode* root){
            vector<int> postorder;
            //Base case :
            if(root == NULL)
                return postorder;
            stack<TreeNode*> s1,s2;
            s1.push(root);
            while(!s1.empty()){
                root=s1.top();
                s1.pop();
                s2.push(root);

                if(root->left != NULL){
                    s1.push(root->left);
                }

                if(root->right != NULL){
                    s1.push(root->right);
                }
            }

            while(!s2.empty()){
                postorder.push_back(s2.top()->val);
                s2.pop();
            }
            return postorder;
        }
};